pkg/utils/replicaresource/utils.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package replicaresource
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	SupportedSchedulerProviders = sets.New("volcano")
+)
+
+const (
+	PodGroupNameFmt = "%s-%s-%s"
+)
+
+// GetPodGroupName returns the name of the PodGroup for a given LeaderWorkloadSet name, group index and revision (leaderworkerset.sigs.k8s.io/template-revision-hash).
+func GetPodGroupName(lwsName, groupIndex, revision string) string {
+	return fmt.Sprintf(PodGroupNameFmt, lwsName, groupIndex, revision)
```

Comment:
Will the name be too long? Will it lead to any problem if we only use `<lwsname>-<groupIndex>`?

Reply:
I used `<lwsname>-<groupIndex>` before. However, during a rolling upgrade, the leader pod will be rebuilt and the podgroup will be rebuilt as well. I am not sure if this will happen:
1. The podgroup is still being deleted
2. The new leader pod has been rebuilt, and the new podgroup also needs to be recreated
3. A podgroup with the same name already exists (it is still being deleted in step 1), so a new podgroup is not created
4. Podgroup deletion completed
5. The new leader pod and worker pods cannot be scheduled because they have no podgroup and remain pending.
6. Rolling update interrupt
If I change to `<lwsname>-<groupIndex>-<revision>`. At least the podgroup will not have the same name before and after the rebuild, so this situation will not happen.

Reply:
Sure, I have added a comment:
`// An example can be like "lws-1-dd6699c7c", where "lws" is the LeaderWorkerSet name, "1" is the group index, and "dd6699c7c" is the revision hash.`